### ISSUE TITLE
TextPrompt.cs "Invalid Input" validation error for empty Nullable<T>

### ISSUE THREAD

**Issue by iraklikhitarishvili** [spectreconsole/spectre.console#998]
**Information**
 - OS: Windows/WSL2
 - Version: 20.04.4 LTS
 - Terminal: Windows Terminal

**Describe the bug**
"Invalid Input" validation error is returned if:
1. prompt is `TextPrompt<Nullable<T>>`
2. `AllowEmpty()` is specified
3. input  is empty 

**To Reproduce**
Code snippet for prompt:
```csharp
var result = AnsiConsole.Prompt(new TextPrompt<int?>("Some message").AllowEmpty());
AnsiConsole.MarkupLine($"Result is {result.Value}");
```

Output:
![invalid_input](https://user-images.githubusercontent.com/7589748/194351588-c1e923da-123c-44ee-a35e-5a3c88fb5390.jpg)

**Expected behavior**
Since :
- `AllowEmpty` is true 
- `TypeConvertor` for `Nullable<T>` returns `true`  

Validation error should not occur.  Instead null should be returned.

**Additional context**
In implementation:  
- Empty input is checked [here](https://github.com/spectreconsole/spectre.console/blob/main/src/Spectre.Console/Prompts/TextPrompt.cs#L131) and then 
- `AllowEmpty` flag is checked [here](https://github.com/spectreconsole/spectre.console/blob/main/src/Spectre.Console/Prompts/TextPrompt.cs#L141)

Only after this following [snippet ](https://github.com/spectreconsole/spectre.console/blob/main/src/Spectre.Console/Prompts/TextPrompt.cs#L163) is executed
```csharp
else if (!TypeConverterHelper.TryConvertFromString<T>(input, out result) || result == null)
{
    console.MarkupLine(ValidationErrorMessage);
    WritePrompt(console);
    continue;
}
```
Will it be wrong to remove `|| result == null`?
`AllowEmpty` is already checked and it's allowed.
`TypeConverterHelper.TryConvertFromString<T>(input, out result)` says that conversion is successful.

**Comment by nils-a** [comment]
Closing as duplicate of #714
